packages/control-property-editor-common/src/api.ts

```diff
@@ -0,0 +1,213 @@
+export const INPUT_EDITOR_TYPE = 'input';
+export const DROPDOWN_EDITOR_TYPE = 'dropdown';
+export const CHECKBOX_EDITOR_TYPE = 'checkbox';
+
+interface ControlPropertyBase<T, V, E> {
+    type: T;
+    editor: E;
+    name: string;
+    readableName: string;
+    value: V | string; // string: expression binding
+    isEnabled: boolean;
+    isIcon?: boolean;
+    label?: string;
+    errorMessage?: string;
+    ui5Type?: string;
+    documentation?: {
+        defaultValue: string;
+        description: string;
+        propertyName: string;
+        type?: string;
+        propertyType?: string;
+    };
+}
+
+export type BooleanControlProperty = ControlPropertyBase<
+    typeof BOOLEAN_VALUE_TYPE,
+    boolean,
+    typeof CHECKBOX_EDITOR_TYPE
+>;
+
+export type IntegerControlProperty = ControlPropertyBase<typeof INTEGER_VALUE_TYPE, number, typeof INPUT_EDITOR_TYPE>;
+
+export type FloatControlProperty = ControlPropertyBase<typeof FLOAT_VALUE_TYPE, number, typeof INPUT_EDITOR_TYPE>;
+
+export type StringControlProperty = ControlPropertyBase<typeof STRING_VALUE_TYPE, string, typeof INPUT_EDITOR_TYPE>;
+
+export interface StringControlPropertyWithOptions
+    extends ControlPropertyBase<typeof STRING_VALUE_TYPE, string, typeof DROPDOWN_EDITOR_TYPE> {
+    options: { key: string; text: string }[];
+}
+
+export type ControlProperty =
+    | BooleanControlProperty
+    | IntegerControlProperty
+    | FloatControlProperty
+    | StringControlProperty
+    | StringControlPropertyWithOptions;
+
+export interface OutlineNode {
+    controlId: string;
+    controlType: string; // as used in rta i.e sap.m.NavContainer
+    name: string;
+    visible: boolean;
+    editable: boolean;
+    children: OutlineNode[];
+}
+
+export interface IconDetails {
+    name: string;
+    content: string;
+    fontFamily: string;
+}
+
+export interface PendingPropertyChange<T extends PropertyValue = PropertyValue> extends PropertyChange<T> {
+    type: 'pending';
+    /**
+     * Indicates if change is before or after current position in undo redo stack
+     */
+    isActive: boolean;
+}
+
+export interface SavedPropertyChange<T extends PropertyValue = PropertyValue> extends PropertyChange<T> {
+    type: 'saved';
+    kind: 'valid';
+    fileName: string;
+    timestamp: number;
+}
+
+export interface UnknownSavedChange {
+    type: 'saved';
+    kind: 'unknown';
+    fileName: string;
+    timestamp?: number;
+}
+export type ValidChange = PendingPropertyChange | SavedPropertyChange;
+export type Change = ValidChange | UnknownSavedChange;
+
+export interface ChangeStackModified {
+    pending: PendingPropertyChange[];
+    saved: SavedPropertyChange[];
+}
+
+export interface PropertyChangeDeletionDetails {
+    controlId: string;
+    propertyName: string;
+    fileName?: string;
+}
+
+/**
+ * ACTIONS
+ */
+
+export interface PayloadAction<T extends string, U> {
+    type: T;
+    payload: U;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
No any please.

Reply:
removed.

Reply:
Thanks, please also remove all other `eslint-disable-next-line` statements in this file (and any other) or add a comment explaining why it is required.

Reply:
removed.